### PR TITLE
Shahin's cleanup

### PR DESCRIPTION
**Pull request by happyhuman**
- Cleaned up README file for setup and setup.sh, per Noah's comments from bugbash.
- Cleaned up the top README.
- Fixed the bug in download-taxi.sh (no more downloading the same thing twice)
- Some other minor fixes.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,24 +10,35 @@ The repository is organized by products:
...
+ -->
 
 ## Getting Started
 
-If this is your first time using [AI Platform](https://cloud.google.com/ml-engine/docs/), we suggest you take a look at the [Introduction to AI Platform](https://cloud.google.com/ml-engine/docs/technical-overview) docs to get started.
+We highly recommend that you start with our [Quick Start Sample](./quickstart).
 
 ## Navigating this Repository
 
-In order to run any of the samples in this repository, you'll need to setup your environment first.
```
Comment (gogasca) [inline README.md]: Does it makes sense to mention other frameworks here?

File: quickstart/scripts/train-local.sh [hunk 2]
```diff
@@ -30,3 +30,5 @@ gcloud ai-platform local train \
         --package-path=${PACKAGE_PATH} \
```
Comment (gogasca) [inline quickstart/scripts/train-local.sh]: Do we need Kokoro test for quick start?

File: setup/README.md [hunk 3]
```diff
@@ -35,43 +35,50 @@ gcloud services enable storage-component.googleapis.com
...
-   ```bash
-     source ./setup.sh
-   ```
+In order to create a service account key, you may follow the instructions [here](https://cloud.google.com/iam/docs/creating-managing-service-account-keys#iam-service-account-keys-create-gcloud).
+Alternatively, you may run this in your terminal, after setting the values for the 3 environment variables:
 
-b) In your terminal copy and paste the following with the corresponding edits.
 
 ```bash
-export RUNTIME_VERSION=1.13
```
Comment (gogasca) [inline setup/README.md]: Why to delete these? These are required.

File: setup/README.md [hunk 4]
```diff
@@ -35,43 +35,50 @@ gcloud services enable storage-component.googleapis.com
...
+#Creathe service account:
+gcloud iam service-accounts create ${MY_SERVICE_ACCOUNT} --display-name="Service Account for ai-platform-samples repo"
 
-# Replace "your-gcp-bucket-name" with a universally unique name for a GCS bucket.
-export BUCKET_NAME="your-gcp-bucket-name"
+# Grant the required roles:
+gcloud projects add-iam-policy-binding endromodal --member serviceAccount:${SERVICE_ACCOUNT_NAME}@${PROJECT_ID}.iam.gserviceaccount.com --role roles/ml.developer
+gcloud projects add-iam-policy-binding endromodal --member serviceAccount:${SERVICE_ACCOUNT_NAME}@${PROJECT_ID}.iam.gserviceaccount.com --role roles/storage.objectAdmin
 
-# Replace "path/to/service/account/key" with the full path to the
```
Comment (gogasca) [inline setup/README.md]: This flow needs to be rearranaged, in item 6 you talk about the credentials and also include the env variables for project/, then in 7 you talk about the env variables.
The way it was it was cleared in terms you have option a or b to setup your env variables, now looks confusing

File: setup/setup.sh [hunk 5]
```diff
@@ -18,38 +18,51 @@
...
 
-
 function export_variables(){
     # Set variables.
     export RUNTIME_VERSION=1.13
     export PYTHON_VERSION=3.5
     export REGION=us-central1
 
-    # Replace "your-gcp-project-id" with your GCP PROJECT ID
-    export PROJECT_ID="your-gcp-project-id"
```
Comment (gogasca) [inline setup/setup.sh]: can we simplify IF and ELSE in a function?

File: quickstart/scripts/train-local.sh (hunk 2 above)
Comment (happyhuman) [reply quickstart/scripts/train-local.sh]: Yes for sure. We will add it in a separate PR (perhaps we can work on it together as our standard for writing unit tests).

File: setup/README.md (hunk 3 above)
Comment (happyhuman) [reply setup/README.md]: I changed setup.sh (and therefore README) a little. Previously, you either had to modify setup.sh and update PROJECT_ID and some other variables and run it; or you needed to do it via console.

The new setup.sh allows for a hybrid solution, where you can modify some or all of the variables in setup.sh and run it, or set the variables in console and then run setup.sh (it is a bit smarter that way). setup.sh prints out all the variables that were set, as well as error messages for variables that are needed to be set.

Another reason for the change was that PROJECT_ID and GOOGLE_APPLICATION_CREDENTIALS were likely to be set in the previous steps of this README (due to Noah's comments in BugBash). In order to unify the setup process, some of these changes were necessary.

File: setup/README.md (hunk 4 above)
Comment (happyhuman) [reply setup/README.md]: Yeah, I agree. Let me see how I can fix this....

File: setup/setup.sh (hunk 5 above)
Comment (happyhuman) [reply setup/setup.sh]: What exactly do you have in mind?

File: README.md (hunk 1 above)
Comment (happyhuman) [reply README.md]: Done.

File: setup/README.md (hunk 3 above)
Comment (gogasca) [reply setup/README.md]: Got it, 

PS: there is a typo on Create.

**Pull request merged**